nx/lib/nx.ex

```diff
@@ -2763,6 +2763,27 @@ defmodule Nx do
     elem(shape, index)
   end
 
+  @doc """
+  Returns the index of the given axis in the tensor.
+
+  ### Examples
+
+      iex> Nx.axis_index(Nx.iota({100, 10, 20}), 0)
+      0
+
+      iex> Nx.axis_index(Nx.iota({100, 10, 20}), -1)
+      2
+
+      iex> Nx.axis_index(Nx.iota({100, 10, 20}, names: [:batch, :x, :y]), :x)
+      1
+
+  """
+  @doc type: :shape
+  def axis_index(tensor, axis) do
```

Comment:
Technically the returned value is also a valid axis, so I'm not sure about the naming. Alternatives: `normalized_axis`, `plain_axis`, wdyt?